### DISCUSSION TITLE
What are the other branches of emscripten for?

### DISCUSSION THREAD

**Discussion by msqr1** [emscripten-core/emscripten#21837] [Q&A] [answered]
Can we delete them?

**Comment by sbc100** [comment, marked as answer]
Most of them represent WIP branches by the team.  In the past we encouraged folks to work upstream rather than in a fork.  However, due the high number of branches (as you have noticed) we recently started discouraging folks from creating branches upstream (here), and we are encouraging folks to clean up their old branches.

Some of the branches are indeed stale and can be removed, but I think I've found all the low hanging fruit already there.  Other branches represent work that is ongoing.   For branches that don't yet have PRs it should be easy enough to simply move them to a fork.  For branches with an open PR I'm not sure its so easy.   For now at least, we are trying to reduce the number of branches incrementally over time rather than try find a way to do it all at once.

**Reply by msqr1** [reply]
So each branch is a PR?

**Reply by sbc100** [reply]
Pretty much, yes.   Or some work in progress that hasn't get been turned into one.
